src/shared/tpm2-util.c

```diff
@@ -1453,3 +1455,24 @@ int tpm2_primary_alg_from_string(const char *alg) {
                 return TPM2_ALG_RSA;
         return -EINVAL;
 }
+
+Tpm2Support tpm2_support(void) {
+        Tpm2Support support = TPM2_SUPPORT_NONE;
```

Comment:
Is it work caching this locally? Result won't change during execution, right?

Reply:
so we already cache the firmware check in efi_has_tpm2(), so we cache *some* thing.

The driver check we don't cache, because initially at boot there's a good chance the drivers aren't loaded yet, but later they will. Hence it's only safe to check each time, and thus not cache.

Hence: i think we already cache as much as makes sense here.